### PR TITLE
add new tests for re_rebalance_coherence

### PR DESCRIPTION
**Pull request by nikomatsakis**
cc https://github.com/rust-lang/rust/issues/63599 and https://github.com/rust-lang/rust/issues/64412

### CODE REVIEW

File: src/test/ui/coherence/impl[t]-foreign[foreign[t],local]-for-foreign.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#![feature(re_rebalance_coherence)]
+
+// run-pass
```
Comment (Centril) [inline src/test/ui/coherence/impl[t]-foreign[foreign[t],local]-for-foreign.rs]: Can you get if the tests in the PR need to be run-pass? Seems check-pass is sufficient?

**Comment by Centril** [comment]
Also, PR builder failed, @bors r-

File: src/test/ui/coherence/impl[t]-foreign[foreign[t],local]-for-foreign.rs (hunk 1 above)
Comment (nikomatsakis) [reply src/test/ui/coherence/impl[t]-foreign[foreign[t],local]-for-foreign.rs]: yep fixed

**Pull request merged**